RNTester/NativeModuleExample/ScreenshotMacOS.mm

```diff
@@ -141,16 +142,71 @@ void TakeScreenshot(
   }
 };
 
-@implementation ScreenshotManagerTurboModuleManagerDelegate
+// winrt code will be move to another files once it is properly implemented
+
+namespace winrt::Microsoft::ReactNative
+{
+
+struct MacOSReactContext : implements<MacOSReactContext, IReactContext>
+{
+  IReactPropertyBag Properties() const noexcept
+  {
+    VerifyElseCrash(false);
+  }
+  
+  IReactNotificationService Notifications() const noexcept
+  {
+    VerifyElseCrash(false);
+  }
+  
+  IReactDispatcher UIDispatcher() const noexcept
+  {
+    VerifyElseCrash(false);
+  }
+  
+  IReactDispatcher JSDispatcher() const noexcept
+  {
+    VerifyElseCrash(false);
+  }
+  
+  void CallJSFunction(const hstring& moduleName, const hstring& methodName, const JSValueArgWriter& paramsArgWriter) noexcept
+  {
+    VerifyElseCrash(false);
+  }
+  
+  void EmitJSEvent(const hstring& eventEmitterName, const hstring& eventName, const JSValueArgWriter& paramsArgWriter) noexcept
+  {
+    VerifyElseCrash(false);
+  }
+};
+
+}
+
+@implementation ScreenshotManagerTurboModuleManagerDelegate {
+  std::shared_ptr<winrt::Microsoft::ReactNative::TurboModulesProvider> provider;
+  winrt::Microsoft::ReactNative::IReactContext reactContext;
+}
 
 - (std::shared_ptr<facebook::react::TurboModule>)
   getTurboModule:(const std::string &)name
   jsInvoker:(std::shared_ptr<facebook::react::CallInvoker>)jsInvoker
 {
-  if (name == "ScreenshotManager")
+  if (!provider)
   {
-    return std::make_shared<ScreenshotManagerTurboModule>(jsInvoker);
+    provider = std::make_shared<winrt::Microsoft::ReactNative::TurboModulesProvider>();
+    reactContext = winrt::make<winrt::Microsoft::ReactNative::MacOSReactContext>();
+    provider->SetReactContext(reactContext);
+    
+    provider->AddModuleProvider(
+      L"ScreenshotManager",
+      winrt::Microsoft::ReactNative::MakeModuleProvider<ScreenshotManagerCxx>()
+      );
   }
+  return provider->getModule(name, jsInvoker);
+  // if (name == "ScreenshotManager")
+  // {
+  //   return std::make_shared<ScreenshotManagerTurboModule>(jsInvoker);
+  // }
   return nullptr;
```

Comment:
This will be deleted in the next pull request, I believe `ScreenshotManagerTurboModule` is able to be completely converted to `ScreenManagerCxx` at that moment.